src/elements/e-video.vue

```diff
@@ -97,108 +85,126 @@
 
       /**
        * Allows to set the video player width.
+       * width can either be responsive or hard coded.
+       * youtube standard -> width="560".
        */
       width: {
-        type: [String, Number],
-        default: 560,
+        type: String,
+        default: null,
       },
 
       /**
        * Allows to set the video player height.
+       * height can either be responsive or hard coded.
+       * youtube standard -> height="315".
        */
       height: {
-        type: [String, Number],
-        default: 315,
-      },
-
-      /**
-       * Defines the players size to be responsive 100% width to the parent container
-       */
-      responsive: {
-        type: Boolean,
-        default: false,
+        type: String,
+        default: null,
       },
     },
 
     // emits: {},
     // setup(): Setup {},
 
     computed: {
+      /**
+       * Returns a validity check for the video url.
+       */
+      validateUrl(): boolean {
+        if (this.videoUrl) {
+          /**
+           * This is for sources with multiple allowed urls.
+           * e.g. youtube (standard, short)
+           */
+          if (Array.isArray(VideoUrlSource[this.source])) {
+            const videoSources = VideoUrlSource[this.source] as string[];
+
+            return this.arrayHasString(videoSources, this.videoUrl);
+          }
+
+          if (!this.videoUrl.startsWith(VideoUrlSource[this.source] as string)) {
+            // eslint-disable-next-line no-console
+            console.error('Invalid video url', this.$el);
+
+            return false;
+          }
+
+          return true;
```

Comment:
todo: this line can be removed. Just do nothing, if everything is OK and use the `return true` at the end of the method.